src/modules/emote_menu/components/Sidebar.jsx

```diff
@@ -1,65 +1,176 @@
+        onClick={() => onClick(category.id)}
+        onKeyDown={() => onClick(category.id)}
+        className={classNames(styles.navItem, {
+          [styles.active]: category.id === section.eventKey,
+        })}>
+        {category.icon}
+      </div>
+    ));
+  }
 
   return (
-    <div className={styles.sidebar} ref={containerRef}>
-      <Nav vertical appearance="subtle">
-        {categories.map((category) => {
-          const isActive = category.id === section.eventKey;
-
-          return (
-            <Whisper
-              key={category.id}
-              placement="right"
-              trigger="hover"
-              delay={200}
-              speaker={<Tooltip>{category.displayName}</Tooltip>}
-              onClick={() => onChange(category.id)}>
-              <Nav.Item
-                active={isActive}
-                icon={<div className={styles.navItemContent}>{category.icon}</div>}
-                className={classNames(styles.navItem, {[styles.active]: isActive})}
-              />
-            </Whisper>
-          );
+    <div className={styles.sidebar}>
+      <div
+        className={classNames(styles.content, {
+          [styles.emojiButtonHidden]: emojiButtonHidden,
+          [styles.emojiButtonVisible]: !emojiButtonHidden,
         })}
-      </Nav>
+        ref={containerRef}>
+        {createCategories(categories.top)}
+        <DragDropContext onDragEnd={({source, destination}) => handleReorder(source.index, destination.index)}>
+          <Droppable droppableId="droppable" type="list" direction="vertical">
+            {(provided) => (
+              <div {...provided.droppableProps} ref={provided.innerRef}>
+                {middleCategories.map((category, index) => (
+                  <Draggable key={category.id} draggableId={category.id} index={index}>
+                    {renderDraggable((providedItem, snapshotItem) => (
+                      <div
+                        role="button"
+                        ref={providedItem.innerRef}
+                        {...providedItem.draggableProps}
+                        {...providedItem.dragHandleProps}
+                        style={providedItem.draggableProps.style}
+                        onClick={() => onClick(category.id)}
+                        onKeyDown={() => onClick(category.id)}
+                        className={classNames(styles.navItem, {
+                          [styles.dragging]: snapshotItem.isDragging,
+                          [styles.active]: category.id === section.eventKey,
+                        })}>
+                        {category.icon}
+                      </div>
+                    ))}
+                  </Draggable>
+                ))}
+                {provided.placeholder}
+              </div>
+            )}
+          </Droppable>
+        </DragDropContext>
+        {createCategories(categories.bottom)}
+      </div>
+      <Divider className={styles.divider} />
+      <div
+        role="button"
```

Comment:
curious: why do we use two buttons for this? could we use 1 button and change the props?